CourseManager/Courses/Course 4 UI and Animations in SwiftUI/Views/Main/HomeViewWeather.swift

```diff
@@ -0,0 +1,112 @@
+    @State var bottomSheetPosition: BottomSheetPosition = .middle
+    @State var bottomSheetTranslation: CGFloat = BottomSheetPosition.middle.rawValue
+    @State var hasDragged: Bool = false
+
+    var bottomSheetTranslationProrated: CGFloat {
+        (bottomSheetTranslation - BottomSheetPosition.middle.rawValue) / (BottomSheetPosition.top.rawValue - BottomSheetPosition.middle.rawValue)
+    }
+
+
+    var body: some View {
+        NavigationView {
+            GeometryReader { geometry in
+                let screenHeight = geometry.size.height + geometry.safeAreaInsets.top + geometry.safeAreaInsets.bottom
+                let imageOffset = screenHeight + 36
+
+                ZStack {
+                    Color.weatherBackground
+                        .ignoresSafeArea()
+                    Image("Background")
+                        .resizable()
+                        .ignoresSafeArea()
+                        .offset(y: -bottomSheetTranslationProrated * imageOffset)
+
+                    Image("House")
+                        .frame(maxHeight: .infinity, alignment: .top)
+                        .padding(.top, 257)
+                        .offset(y: -bottomSheetTranslationProrated * imageOffset)
+
+                    VStack(spacing: -10 * (1 - bottomSheetTranslationProrated)) {
+                        Text("Montreal")
+                            .font(.largeTitle)
+
+                        VStack {
+                            Text(attributedString)
+                            Text("H:24°   L:18°")
+                                .font(.title3.weight(.semibold))
+                                .opacity(1 - bottomSheetTranslationProrated)
+
+                        }
+
+                        Spacer()
+                    }
+                    .padding(.top, 51)
+                    .offset(y: -bottomSheetTranslationProrated * 46)
+
+                    BottomSheetView(position: $bottomSheetPosition) {
+                    } content: {
+                        ForecastView(bottomSheetTranslationProrated: bottomSheetTranslationProrated)
+                    }
+                    .onBottomSheetDrag { translation in
+                        bottomSheetTranslation = translation / screenHeight
+                        withAnimation(.easeInOut) {
+                               if bottomSheetPosition == BottomSheetPosition.top {
+                                   hasDragged = true
+                               } else {
+                                   hasDragged = false
+                               }
+                           }
+                    }
+                    TabBarWeather(action: {
+                        bottomSheetPosition = .top
+                    })
+                    .offset(y: bottomSheetTranslationProrated * 115)
+
+                }
+            }
+            .navigationBarBackButtonHidden(true)
+        }
+    }
+    private var attributedString: AttributedString {
```

Comment:
Fix alignment here

Reply:
Also the point to consider, since the value is dependent only on the change of `hasDragged`, it may be a waste of resources to recompute it on every view update. Obviously, it isn't the issue here, but it may be in other place

Reply:
Ok got it, will keep in mind.